custom_model_runner/datarobot_drum/drum/drum.py

```diff
@@ -759,26 +775,65 @@ def _run_inside_docker(self, options, run_mode, raw_arguments):
 
     def _maybe_build_image(self, docker_image_or_directory):
         ret_docker_image = None
-        client = docker.client.from_env()
 
         if os.path.isdir(docker_image_or_directory):
             docker_image_or_directory = os.path.abspath(docker_image_or_directory)
-            self.logger.info(
-                "Building a docker image from directory {}...".format(docker_image_or_directory)
+            docker_build_msg = "Building a docker image from directory: {}...".format(
+                docker_image_or_directory
             )
+            self.logger.info(docker_build_msg)
             self.logger.info("This may take some time")
             try:
-                image, _ = client.images.build(path=docker_image_or_directory)
-                ret_docker_image = image.id
-            except docker.errors.BuildError as e:
-                self.logger.error("Hey something went wrong with image build!")
-                for line in e.build_log:
+                # Set image tag to the dirname of the docker context.
+                # This may be confusing if try to build different images from different contexts,
+                # with the same folder name:
+                # /path1/my_env
+                # /path2/my_env
+                #
+                # If image with the tag `my_env` exists, it will be untagged.
```

Comment:
why this

Reply:
It's nice to tag the image, and the easiest way is to take dir name.

I could add a param, `--tag` or `--docker-tag` which will be used if dir path is provided with `--docker`, but i'd hold on until more feedback.